src/lib/support/DefaultStorageKeyAllocator.h

```diff
@@ -40,7 +40,7 @@ class DefaultStorageKeyAllocator
     // Access Control List
 
     const char * AccessControlList() { return Format("acl"); }
-    const char * AccessControlEntry(size_t index) { return Format("acl/%zx", index); }
+    const char * AccessControlEntry(size_t index) { return Format("acl/%x", static_cast<unsigned int>(index)); }
```

Comment:
Can we add a static_assert that sizeof(unsigned int) >= sizeof(size_t)?

Or are we not really worried about overflow on the cast?

Reply:
We are really not worried at all, because it's always going to be >= 65535 (and we don't support 8/16 bit platforms...) and really, it's going to be 32 or 64 bits. If you are on an embedded target, and have 65535 ACL entries, there is going to be a boom. If there are 2^32-1,  worse still ;)

Reply:
OK, I buy that.  It might be worth a comment explaining why the cast is OK at this specific spot.

Reply:
I've added a comment.